mm/ktsan/trace.c

```diff
@@ -86,21 +96,22 @@ void kt_trace_restore_stack(kt_thr_t *thr, kt_time_t clock, kt_stack_t *stack)
 	kt_spin_lock(&trace->lock);
 
 	if (header->clock > clock) {
-		stack->size = 0;
+		state->stack.size = 0;
+		state->cpu_id = -1;
 		kt_spin_unlock(&trace->lock);
 		return;
 	}
 
-	memcpy(stack, &header->stack, sizeof(*stack));
+	memcpy(state, &header->state, sizeof(*state));
 	end = clock % KT_TRACE_SIZE;
 	beg = round_down(end, KT_TRACE_PART_SIZE);
-	kt_trace_follow(trace, beg, end, stack);
+	kt_trace_follow(trace, beg, end, state);
 
 	event = &trace->events[end];
 	if (event->type == kt_event_type_mop) {
-		BUG_ON(stack->size + 1 == KT_MAX_STACK_FRAMES);
-		stack->pc[stack->size] = event->pc;
-		stack->size++;
+		BUG_ON(state->stack.size + 1 == KT_MAX_STACK_FRAMES);
```

Comment:
Why do we do this only for kt_event_type_mop? We write pc for lots of events, and it seems to me that any event that is not func_entry/exit needs to add top pc.